packages/contracts/test/token/erc20/governance-erc20.ts

```diff
@@ -0,0 +1,221 @@
+
+      const mintAmount = 100;
+      token.mint(receiverAddr, mintAmount);
+
+      expect(await token.balanceOf(receiverAddr)).to.eq(
+        oldBalance.add(mintAmount)
+      );
+    });
+  });
+
+  describe('delegate', async () => {
+    beforeEach(async function () {
+      token = await GovernanceERC20.deploy(...defaultInitData);
+    });
+
+    it('delegates voting power to another account', async () => {
+      const balanceSigner0 = await token.balanceOf(signers[0].address);
+      const balanceSigner1 = await token.balanceOf(signers[1].address);
+
+      // delegate the votes of signers[0] to signers[1]
+      await token.connect(signers[0]).delegate(signers[1].address);
+
+      // signers[0] delegates to signers[1]
+      expect(await token.delegates(signers[0].address)).to.eq(
+        signers[1].address
+      );
+
+      // signers[1] delegates to herself
+      expect(await token.delegates(signers[1].address)).to.eq(
+        signers[1].address
+      );
+
+      // balances remain the same
+      expect(await token.balanceOf(signers[0].address)).to.eq(balanceSigner0);
+      expect(await token.balanceOf(signers[1].address)).to.eq(balanceSigner1);
+
+      // the voting power changes
+      expect(await token.getVotes(signers[0].address)).to.eq(0);
+      expect(await token.getVotes(signers[1].address)).to.eq(
+        balanceSigner1.add(balanceSigner0)
+      );
+    });
+
+    it('is checkpointed', async () => {
+      const balanceSigner0 = await token.balanceOf(signers[0].address);
+      const balanceSigner1 = await token.balanceOf(signers[1].address);
+      const balanceSigner2 = await token.balanceOf(signers[2].address);
+
+      let tx1 = await token.connect(signers[0]).delegate(signers[1].address);
+      await ethers.provider.send('evm_mine', []);
+
+      // verify that current votes are correct
+      expect(await token.getVotes(signers[0].address)).to.eq(0);
+      expect(await token.getVotes(signers[1].address)).to.eq(
+        balanceSigner1.add(balanceSigner0)
+      );
+      expect(await token.getVotes(signers[2].address)).to.eq(balanceSigner2);
+
+      let tx2 = await token.connect(signers[0]).delegate(signers[2].address);
+      await ethers.provider.send('evm_mine', []);
+
+      // verify that current votes are correct
+      expect(await token.getVotes(signers[0].address)).to.eq(0);
+      expect(await token.getVotes(signers[1].address)).to.eq(balanceSigner1);
+      expect(await token.getVotes(signers[2].address)).to.eq(
+        balanceSigner2.add(balanceSigner0)
+      );
+
+      // verify that past votes are correct
+      expect(
+        await token.getPastVotes(signers[0].address, tx1.blockNumber)
+      ).to.eq(0);
+      expect(
+        await token.getPastVotes(signers[1].address, tx1.blockNumber)
+      ).to.eq(balanceSigner1.add(balanceSigner0));
+      expect(
+        await token.getPastVotes(signers[2].address, tx1.blockNumber)
+      ).to.eq(balanceSigner2);
```

Comment:
I'd add one more test that after minting, users automatically have the feature delegated. This is because: 

```
if (from == address(0) && to != address(0) && delegates(to) == address(0)) {
            _delegate(to, to);
        }
```

So, we need to be testing this. Then, one more test(or inside the same test, try to transfer some tokens to someone who have never minted before, then check that it has 0 delegating power, then mint from it, and suddently, this man would have `transferAmount` + `mintAmount` as a delegated power.

One more test would be to check that `_delegate` only gets called one time only (first time of minting). but this would require smock. I'd do it if I were you.

Reply:
Agreed with @novaknole to be done in another task